subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -56,7 +56,15 @@
     /**
      * Executes the work synchronously.
      */
-    WorkOutput execute(@Nullable InputChangesInternal inputChanges, InputChangesContext context);
+    WorkOutput execute(ExecutionRequest executionRequest);
+
+    interface ExecutionRequest {
```

Comment:
I don't like `ExecutionRequest` much. `ExecutionRequest` sounds like something we would pass to the execution engine, not to the unit of work. Though I don't have a much better name for it now. The only thing I can come up with is something like `ExecutionContext`. So you can leave it as is if you want to.

Reply:
That was my thinking process, too. :/

Reply:
I also feel that `Request` sounds wrong. As much as I don't like the generic `Context` thing I think it better represents what this is, so I'd use `ExecutionContext` instead. Alternatively: `ExecutionDetails`, `ExecutionEnvironment`.

Reply:
`ExecutionParameters` and `UnitOfWorkParameters`  come to mind.

Reply:
Thanks for all the feedback! I agree that "request" is far from perfect. I'd like to reserve `*Context` for the internal thingies from now on. I don't think "details" or "environment" captures what's going on here well. And we already use "parameters" in the context of `WorkParameters` and `TransformParameters` which I feel would be confusing.

Perhaps we need a more thorough overhaul of the naming here. I would also like to somehow reduce the amount of "context" we expose by the end of the pipeline: right now almost every context type extends all of its preceding context types, which is definitely not what we need, and adds a lot of noise in implementing these types.

If any of you wants to set up a meeting about better naming here, I'd be very happy to discuss.